content/protocols/sse.textile

```diff
@@ -0,0 +1,230 @@
+    // ... normal message processing
+  }
+
+  // handle connection errors
+  eventSource.onerror = msg => {
+    const err = JSON.parse(msg.data);
+    const isTokenErr = err.code >= 40140 && err.code < 40150;
+    if(isTokenErr) {
+      eventSource.close();
+      connectToAbly();
+    } else {
+      // ... handle other types of error -- for example, retry on 5xxxx, close on 4xxxx
+    }
+  }
+}
+
+connectToAbly();
+```
+
+An important thing to note here is that the EventSource API tries to auto-reconnect and re-subscribe to the SSE endpoint when any error occurs, even the token expiry error like in this case. This means that upon manually re-subscribing to the SSE endpoint with a new token, there will be two active subscriptions to the endpoint - one with the old token which would continue to throw an error due to expired credentials and another with the new token. Hence, it is important to close the previous @EventSource@ subscription using @eventSource.close()@ before re-subscribing with the new token as shown in the snippet above.
+
+You can take a look at a "demo app":https://sse-token-auth.glitch.me and "a complete code example for implementing message continuity in an SSE subscription when using token auth":https://glitch.com/edit/#!/sse-token-auth.
+
+h2(#channel-options). Channel options
+
+In an SSE connection, it is also possible to specify "channel options":/channels/options as a query string in the connection URL, instead of as a qualifier on an individual channel name. In this case, the given channel options apply to all channel attachments associated with that connection.
+
+Currently both "rewind":/channels/options/rewind and "delta":/channels/options/delta channel options are supported.
+
+h3(#delta-sse). Delta with SSE
+
+If subscribing to a channel in delta mode using SSE then you will need to decode any received delta messages yourself.
+
+When subscribing without an Ably library, the channel @delta@ parameter must be specified using a qualified channel name. For the client libraries that do not currently expose the API, or transports that do not involve using an Ably library, a set of channel parameters can be expressed by including a query string with standard URL query syntax and encoding, within the qualifier part of a channel name. The qualifier part is in square brackets at the start of the channel name.
+
+To specify the parameter @foo@ with value @bar@ on channel @baz@, the qualified channel name would be @[?foo=bar]baz@. If the channel name already has a qualifier, such as @[meta]log@, then the query string follows the existing qualifier, as in @[meta?foo=bar]log@.
+
+Using this syntax with a non-supported Ably library means that channel parameters are specified for the lifetime of the Channel instance; in order to reference the same channel, but with different channel parameters, it is necessary to get a new Channel instance, using a qualified name that includes the new channel parameters.
```

Comment:
Same comment as [here](https://github.com/ably/docs/pull/1784#discussion_r1081569586) if that turns out to be correct 🙂